src/core/dbus-unit.c

```diff
@@ -2080,6 +2081,30 @@ static int bus_unit_set_live_property(
         /* Handles setting properties both "live" (i.e. at any time during runtime), and during creation (for
          * transient units that are being created). */
 
+        if (streq(name, "DebugWait")) {
+                bool d;
+
+                r = sd_bus_message_read(message, "b", &d);
+                if (r < 0)
+                        return r;
+
+                if (SERVICE(u)->exec_command[SERVICE_EXEC_START_PRE])
```

Comment:
Doing the check here means the client will be left hanging.

Instead, do this check in the client: query the properties, and check that `Type` != `oneshot`, and also check that StartPre/StartPost/StopPost and ExecCondition are all empty, and refuse with a clear log message otherwise